### DISCUSSION TITLE
Why does our JWT token still validate even after changing the last digit?

### DISCUSSION THREAD

**Discussion by amarantmeida** [ant-media/Ant-Media-Server#6397] [Q&A] [answered]
Changing the last digit on our JWT token is still validating the token, is this as per design?

**Comment by yashtandon113** [comment, marked as answer]
Hi @amarantmeida 
Thanks for asking the question.

For example, this is the generated JWT token.

```
eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdHJlYW1JZCI6Imhsc2RlbW8iLCJ0eXBlIjoicHVibGlzaCIsImV4cCI6OTk5OTk5OTk5OX0.gvckkkh5zPp8ZWfMHue8xm6_dwCAhvqVnrq7Nua46d4
```

Now if we change the last digit to 5, 6, or 7, it still validates. But if I make it to 3 or 8, then it becomes invalid.

This is an expected behaviour of JWT’s token generation algorithm. Check out the article below for more details.

https://medium.com/@mark_huber/decoding-the-jwt-anomaly-when-changing-a-tokens-last-character-doesn-t-break-verification-d6ab68627afb
